java/vector/src/main/java/org/apache/arrow/vector/NullableVarBinaryVector.java

```diff
@@ -152,12 +135,9 @@ public void get(int index, NullableVarBinaryHolder holder){
          holder.isSet = 0;
          return;
       }
-      final int startOffset = getstartOffset(index);
-      final int dataLength =
-              offsetBuffer.getInt((index + 1) * OFFSET_WIDTH) - startOffset;
       holder.isSet = 1;
-      holder.start = startOffset;
-      holder.end = dataLength;
+      holder.start = getstartOffset(index);
```

Comment:
I think "offset" is well defined in format, so this method can probably just be:

```
getOffset(int index)
```

Reply:
or just:
```
offsetBuffer.getInt(index * OFFSET_WIDTH)
``

Probably it consist with `holder.end = ...`

Reply:
startOffset implies the starting point of VARCHAR element in the data buffer.

Reply:
Ok